include/nlohmann/json.hpp

```diff
@@ -3816,6 +3816,19 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         JSON_THROW(type_error::create(305, "cannot use operator[] with a string argument with " + std::string(type_name()), *this));
     }
 
+    // see https://github.com/nlohmann/json/pull/2685#issuecomment-994015092
+    template<typename T, std::size_t n>
+    reference operator[](T * (&key)[n])
```

Comment:
a compile-time string literal is, i believe, never a non-const reference to char.

and this appears to be a pointer to non-const reference to array of T ?

If it helps you any, you can make an "Alias" typedef, like this:

template<typename T>
using Alias = T;

Then define the signature of your functions like

template<typename T, std::size_t n>
reference operator[](Alias<T[n]> const&)
{
    return ....
}

Reply:
I'm afraid I do not understand what you mean. In particular, what should go to the `....` part.

Reply:
What would go into the "...." part is whatever the operator[] function should do when given a const-ref to char array.

I just don't see how the function as currently written could possible do anything. It's parameter is a reference to array of pointers, not a reference to array of values.

Reply:
https://github.com/nlohmann/json/issues/171#issuecomment-173933616

Looks like I might have misdirected you a bit.  The functions that were added in this commit were the single argument versions:

    template<typename T>
    reference operator[](T* key)
 
    template<typename T>
    const_reference operator[](T* key) const
 
and it was to support string literals converted to plain pointers, not the actual literals themselves:

    const char* _VAR1 = "MyKey";
    j[_VAR1] = 10;

    char* _VAR1 = "MyKey";
    j[_VAR1] = 10;